src/presign/record.rs

```diff
@@ -82,35 +100,39 @@ impl TryFrom<RecordPair> for PresignRecord {
 
         Ok(PresignRecord {
             R,
-            k: private.k.clone(),
+            k: bn_to_scalar(&private.k)?,
             chi: private.chi,
         })
     }
 }
 
 impl PresignRecord {
-    fn x_from_point(p: &CurvePoint) -> Result<k256::Scalar> {
-        let r = &p.0.to_affine().x();
-        Option::from(k256::Scalar::from_repr(*r)).ok_or_else(|| {
+    /// Compute the x-projection of the provided [`CurvePoint`].
+    fn x_from_point(point: CurvePoint) -> Result<Scalar> {
+        let r = &point.0.to_affine().x();
+        Option::from(Scalar::from_repr(*r)).ok_or_else(|| {
             error!("Unable to create Scalar from bytes representation");
             InternalInvariantFailed
         })
     }
 
-    /// Generate a signature share on the given `digest` with
-    /// the [`PresignRecord`].
+    /// Generate a signature share for the given [`Sha256`] instance.
+    ///
+    /// This method consumes the [`PresignRecord`] since it must only be used
+    /// for a single signature.
+    // XXX Replace `Sha256` with `Digest` to make the API clearer?
```

Comment:
Ah, `Digest` would be the output of `Sha256::finalize`. And actually, would a cleaner API be to take a `&[u8]` containing the message, and have `sign` handle the hashing? That avoids the caller needing to think about hashing at all?

Reply:
Oh yeah, that might be nicest. Our spec currently has a whole message to get signed, not the hash itself, so it's probably nicer to just take those message bytes directly.

Reply:
I'll leave this as a separate issue for now, as it'll require writing a wrapper for the `ecdsa::VerificationKey` type (since `VerificationKey::verify_digest` assumes a `Sha256` instance is provided).

Reply:
Made #363 to address this.